Assets/GLTF/Scripts/GLTFLoader.cs

```diff
@@ -191,29 +198,123 @@ protected virtual GameObject CreateNode(Node node)
 				{
-					var childObj = CreateNode(child.Value);
+					var childObj = CreateNode(child);
 					childObj.transform.SetParent(nodeObj.transform, false);
 				}
 			}
 
 			return nodeObj;
 		}
 
-		protected virtual void CreateMeshObject(Mesh mesh, Transform parent)
+		/// <summary>
+		/// Generate Animation components from glTF animations, and attach to game objects
+		/// </summary>
+		protected void AddAnimationToScene(Scene scene, GameObject sceneObj, Animation animation)
 		{
-			foreach (var primitive in mesh.Primitives)
+			// create the animation clip that will contain animation data
+			AnimationClip clip = new AnimationClip();
+			clip.name = animation.Name ?? "GLTFAnimation";
+
+			// needed because Animator component is unavailable at runtime
+			clip.legacy = true;
+
+			foreach (var channel in animation.Channels)
+			{
+				AnimationCurve[] curves = channel.AsAnimationCurves();
+
+				// TODO Memoize
+				// Map the target Node ID to a GameObject, get its name, and prepend parents' names to generate a path.
+				GameObject node = _nodeMap[channel.Target.Node.Id];
+				string nodePath = "";
+				do
+				{
+					nodePath = node.name + (nodePath == "" ? nodePath : "/" + nodePath);
+					node = node.transform.parent.gameObject;
+				} while (node != null && node != sceneObj);
+
+				if (channel.Target.Path == GLTFAnimationChannelPath.translation)
+				{
+					clip.SetCurve(nodePath, typeof(Transform), "localPosition.x", curves[0]);
+					clip.SetCurve(nodePath, typeof(Transform), "localPosition.y", curves[1]);
+					clip.SetCurve(nodePath, typeof(Transform), "localPosition.z", curves[2]);
+				}
+				else if (channel.Target.Path == GLTFAnimationChannelPath.rotation)
+				{
+					clip.SetCurve(nodePath, typeof(Transform), "localRotation.x", curves[0]);
+					clip.SetCurve(nodePath, typeof(Transform), "localRotation.y", curves[1]);
+					clip.SetCurve(nodePath, typeof(Transform), "localRotation.z", curves[2]);
+					clip.SetCurve(nodePath, typeof(Transform), "localRotation.w", curves[3]);
+				}
+				else if (channel.Target.Path == GLTFAnimationChannelPath.scale)
+				{
+					clip.SetCurve(nodePath, typeof(Transform), "localScale.x", curves[0]);
+					clip.SetCurve(nodePath, typeof(Transform), "localScale.y", curves[1]);
+					clip.SetCurve(nodePath, typeof(Transform), "localScale.z", curves[2]);
+				}
+				else if (channel.Target.Path == GLTFAnimationChannelPath.weights)
+				{
+					//TODO: Actually Get Primitive Object Path
+					var primitives = channel.Target.Node.Value.Mesh.Value.Primitives;
+					var targetCount = primitives[0].Targets.Count;
+					for(int primitiveIndex = 0; primitiveIndex < primitives.Count; primitiveIndex++)
+					{
+						string primitiveObjPath = nodePath + "/Primitive" + primitiveIndex;
+						for (int targetIndex = 0; targetIndex < targetCount; targetIndex++)
+						{
+							clip.SetCurve(primitiveObjPath, typeof(SkinnedMeshRenderer), "blendShape."+targetIndex, curves[targetIndex]);
```

Comment:
Nit spaces around binary operators.

Reply:
what does "nit" mean?

Reply:
It's like a nitpick, a small style note that isn't very important.

Reply:
Oh! Yeah I like the spaces as well. It looks great!